### PR TITLE
Added engine version argument to edit command. 

### LINKED ISSUE #13609: Need to specify engine version in azdata command when not using postgresql-12
When trying to edit postgres instance with engine version of 11 it would fail saying: 

pg11bug

To resolve this we could specify --engine-version in the commands. 

- Azure Data Studio Version: 1.25
- Azure Data CLI 0.4.1

Steps to Reproduce:
1. Select PostgreSQL instance with engine version 11
2. Edit a instance on Compute & Storage page
3. View progress notifications

### PR DESCRIPTION
**Pull request by nasc17**
This PR fixes #13609 

When pressing save, edit command takes engine version to add as an azdata argument.

### CODE REVIEW

**Review by Charles-Gagnon — APPROVED** [review]

**Review by Charles-Gagnon — CHANGES_REQUESTED** [review]
Looks like you have some compile errors with your change

**Review by Charles-Gagnon — APPROVED** [review]

**Review by alanrenmsft — APPROVED** [review]

File: extensions/azdata/src/azdata.ts [hunk 1]
```diff
@@ -118,6 +118,7 @@ export class AzdataTool implements azdataExt.IAzdataApi {
 						replaceEngineSettings?: boolean,
 						workers?: number
 					},
+					engineVersion?: string,
```
Comment (ranasaria) [inline extensions/azdata/src/azdata.ts:121]: this should be a parameter within the args object

**Review by ranasaria — CHANGES_REQUESTED** [review]
For consistency, make engineVersion a parameter within the args argument of the edit command. Looks good otherwise.

File: extensions/azdata/src/azdata.ts (hunk 1 above)
Comment (Charles-Gagnon) [reply extensions/azdata/src/azdata.ts:121]: I don't agree with that - and honestly I'm not even sure why we have the args object anyways. That's not how we do the other commands such as dc create. 

Name is another argument but it's at the top level which seems fine to me.

Comment (ranasaria) [reply extensions/azdata/src/azdata.ts:121]: True the args object can be eliminated completely but with it present it makes for confusing usage as to what's in the args and what is not. We should clean this up.

**Pull request merged**